simple/stats/runner.py

```diff
@@ -57,62 +48,69 @@ class Runner:
-    self.special_handlers: dict[str, FileHandler] = {}
+    self.special_files: dict[str, File] = {}
     self.svg_specialized_names: ParentSVG2ChildSpecializedNames = {}
 
-    # Config file driven.
-    if config_file:
-      config_fh = create_file_handler(config_file, is_dir=False)
-      if not config_fh.exists():
-        raise FileNotFoundError("Config file must be provided.")
-      self.config = Config(data=json.loads(config_fh.read_string()))
+    # Config file driven (input paths pulled from config)
+    if config_file_path:
+      with create_store(config_file_path) as config_store:
+        config_data = config_store.as_file().read()
+        self.config = Config(data=json.loads(config_data))
 
       input_urls = self.config.data_download_urls()
       if not input_urls:
         raise ValueError("Data Download URLs not found in config.")
       for input_url in input_urls:
-        self.input_handlers.append(create_file_handler(input_url, is_dir=True))
+        input_store = create_store(input_url)
+        self.stores.append(input_store)
+        self.input_stores.append(input_store)
 
-    #Input dir driven.
+    # Input dir driven (config file found in input dir)
     else:
-      input_dir_fh = create_file_handler(input_dir, is_dir=True)
-      if not input_dir_fh.isdir:
-        raise NotADirectoryError(
-            f"Input path must be a directory: {input_dir}. If it is a GCS path, ensure it ends with a '/'."
-        )
-      self.input_handlers.append(input_dir_fh)
+      input_store = create_store(input_dir_path)
+      self.stores.append(input_store)
+      self.input_stores.append(input_store)
 
-      config_fh = input_dir_fh.make_file(constants.CONFIG_JSON_FILE_NAME)
-      if not config_fh.exists():
-        raise FileNotFoundError("Config file must be provided.")
-      self.config = Config(data=json.loads(config_fh.read_string()))
+      config_file = input_store.as_dir().open_file(
+          constants.CONFIG_JSON_FILE_NAME, create_if_missing=False)
+      self.config = Config(data=json.loads(config_file.read()))
 
-    self.special_file_to_type = self.config.special_files()
+    # Get dict of special file type string to special file name.
+    # Example entry: verticalSpecsFile -> vertical_specs.json
+    self.special_file_names_by_type = self.config.special_files()
 
     # Output directories
-    self.output_dir_fh = create_file_handler(output_dir, is_dir=True)
-    self.nl_dir_fh = self.output_dir_fh.make_file(f"{constants.NL_DIR_NAME}/")
-    self.process_dir_fh = self.output_dir_fh.make_file(
-        f"{constants.PROCESS_DIR_NAME}/")
-
-    self.output_dir_fh.make_dirs()
-    self.nl_dir_fh.make_dirs()
-    self.process_dir_fh.make_dirs()
+    output_store = create_store(output_dir_path, create_if_missing=True)
+    if self.include_input_subdirs:
+      for input_store in self.input_stores:
+        _check_not_overlapping(input_store, output_store)
```

Comment:
Nice check! Perhaps log a warning that they should keep input and output dirs separate.